xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp

```diff
@@ -452,20 +507,177 @@ bool CGUIDialogVideoManagerVersions::ProcessVideoVersion(VideoDbContentType item
+            m_database.DeleteMovie(idMedia);
+          }
+          else
+            return false;
+        }
+      }
+      else
+        m_database.RemoveVideoVersion(idFile);
+    }
+
+    CFileItem item{path, false};
+
+    if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+            CSettings::SETTING_MYVIDEOS_EXTRACTFLAGS))
+    {
+      CDVDFileInfo::GetFileStreamDetails(&item);
+      CLog::LogF(LOGDEBUG, "Extracted filestream details from video file {}",
+                 CURL::GetRedacted(item.GetPath()));
     }
+
+    const int idNewVideoVersion{ChooseVideoAsset(m_videoAsset, VideoAssetType::VERSION)};
+    if (idNewVideoVersion != -1)
+      m_database.AddPrimaryVideoVersion(itemType, dbId, idNewVideoVersion, item);
+
+    return true;
   }
+  return false;
+}
+
+bool CGUIDialogVideoManagerVersions::GetSimilarMovies(const std::shared_ptr<CFileItem>& item,
+                                                      CFileItemList& list,
+                                                      CVideoDatabase& videoDb)
+{
+  list.Clear();
+
+  videoDb.GetSameVideoItems(*item, list);
 
-  // decorate the items
-  for (const auto& item : list)
+  if (list.Size() < 2)
+  {
+    list.Clear();
+    return true;
+  }
+
+  if (!PostProcessList(list, item->GetVideoInfoTag()->m_iDbId))
+    return false;
+
+  return true;
+}
+
+bool CGUIDialogVideoManagerVersions::AddSimilarMovieAsVersion(
+    const std::shared_ptr<CFileItem> itemMovie)
+{
+  // A movie with versions cannot be turned into a version
+  if (itemMovie->GetVideoInfoTag()->HasVideoVersions())
   {
-    item->SetLabel2(item->GetVideoInfoTag()->m_strFileNameAndPath);
+    CGUIDialogOK::ShowAndGetInput(CVariant{40005}, CVariant{40006});
+    return false;
   }
 
-  return ChooseVideoAndConvertToVideoVersion(list, itemType, mediaType, dbId, videodb);
+  // choose a video version type for the video
+  const int idVideoVersion{ChooseVideoAsset(itemMovie, VideoAssetType::VERSION)};
+  if (idVideoVersion < 0)
+    return false;
+
+  CVideoDatabase videoDb;
+  if (!videoDb.Open())
+  {
+    CLog::LogF(LOGERROR, "Failed to open video database!");
+    return false;
+  }
+
+  const int sourceDbId{itemMovie->GetVideoInfoTag()->m_iDbId};
+  const int targetDbId{m_videoAsset->GetVideoInfoTag()->m_iDbId};
+  return videoDb.ConvertVideoToVersion(VideoDbContentType::MOVIES, sourceDbId, targetDbId,
+                                       idVideoVersion);
+}
+
+bool CGUIDialogVideoManagerVersions::PostProcessList(CFileItemList& list, int dbId)
+{
+  // Exclude the provided dbId and decorate the items
+
+  int i = 0;
+  while (i < list.Size())
+  {
+    const auto tag{list[i]->GetVideoInfoTag()};
+    if (tag->m_iDbId == dbId)
+    {
+      list.Remove(i);
+      // i is not incremented for the next iteration because the removal shifted what would have
+      // been the next item into the current position.
+      continue;
+    }
+
+    list[i]->SetLabel2(tag->m_strFileNameAndPath);
```

Comment:
`list[i]`is used twice in this loop. Looking at the implementation of `CFileItemList::operator[]`, a variable seems to be valuable here.

```c++
CFileItemPtr CFileItemList::operator[] (int iItem)
{
  return Get(iItem);
}

CFileItemPtr CFileItemList::Get(int iItem) const
{
  std::unique_lock<CCriticalSection> lock(m_lock);

  if (iItem > -1 && iItem < (int)m_items.size())
    return m_items[iItem];

  return CFileItemPtr();
}
``

Reply:
I didn't notice, ok.